File: create-web-service/src/main/java/com/william/createwebservice/service/UserService.java
package com.william.createwebservice.service;

import com.william.createwebservice.shared.dto.UserDTO;
import org.springframework.security.core.userdetails.UserDetailsService;

import java.util.List;
import java.util.Set;

public interface UserService extends UserDetailsService {
    List<UserDTO> getUsers(int page, int limit);
    List<UserDTO> getUsersByRoleName(String roleName);
    UserDTO getUser(String email);
    UserDTO getUserByUserId(String userId);
    UserDTO createUser(UserDTO user);
    UserDTO updateUser(String userId, UserDTO user);
    void deleteUser(String userId);
    Boolean existsByEmail(String email);
}

File: react-app/src/services/UserService.js
import axios from "axios";

const USER_API_BASE_URL = "http://localhost:8082/create-web-service/api/users";

class UserService {
    getUserByUserId(userId) {
        return axios.get(USER_API_BASE_URL + "/" + userId);
    }

    getUserByUserRoleName(roleName) {
        return axios.get(USER_API_BASE_URL + "/roles/"+ roleName);
    }

    getUsers(page, limit) {
        return axios.get(USER_API_BASE_URL + "?page=" + page + "&lmit=" +limit);
    }

    addUser(userData){
        return axios.post(USER_API_BASE_URL, userData);
    }

    deleteUser(userId){
        return axios.delete(USER_API_BASE_URL + "/" + userId);
    }

    updateUser(userData){
        return axios.put(USER_API_BASE_URL + '/' + userData.userId, userData);
    }
}

export default new UserService();